src/main/java/com/feywild/feywild/block/MagicalBrazierBlock.java

```diff
@@ -0,0 +1,141 @@
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.item.enchantment.EnchantmentHelper;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.RenderShape;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
+import net.minecraft.world.level.material.Material;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.VoxelShape;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+
+import javax.annotation.Nonnull;
+import java.util.Random;
+import java.util.function.Consumer;
+import java.util.function.ToIntFunction;
+
+public class MagicalBrazierBlock extends BlockBE<MagicalBrazier> {
+
+    public static final BooleanProperty BRAZIER_LIT = BooleanProperty.create("brazier_lit");
+    protected static final VoxelShape SHAPE = Block.box(0.0D, 7.0D, 0.0D, 16.0D, 14.0D, 16.0D);
+    private final int FIRE_DAMAGE = 3;
+    private boolean spawnParticles;
+
+    public MagicalBrazierBlock(ModX mod) {
+        super(mod, MagicalBrazier.class, Properties.of(Material.METAL)
+                .strength(0f)
+                .lightLevel(litBlockEmission())
+                .noOcclusion());
+
+        this.spawnParticles = false;
+        this.registerDefaultState(getStateDefinition().any().setValue(BRAZIER_LIT, false));
+    }
+
+    public static boolean isLit(BlockState state) {
+        return state.hasProperty(BRAZIER_LIT) && state.getValue(BRAZIER_LIT);
+    }
+
+    public static void dowse(Level level, BlockPos pos, BlockState state) {
+        level.playSound(null, pos, SoundEvents.GENERIC_EXTINGUISH_FIRE, SoundSource.BLOCKS, 1.0F, 1.0F);
+    }
+
+    private static ToIntFunction<BlockState> litBlockEmission() {
+        return (brazier) -> brazier.getValue(BRAZIER_LIT) ? 10 : 0;
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public void entityInside(@Nonnull BlockState state, @Nonnull Level level, @Nonnull BlockPos pos, Entity entity) {
+        if (!entity.fireImmune() && state.getValue(BRAZIER_LIT) && entity instanceof LivingEntity && !EnchantmentHelper.hasFrostWalker((LivingEntity) entity)) {
+            entity.hurt(DamageSource.IN_FIRE, (float) this.FIRE_DAMAGE);
+        }
+        super.entityInside(state, level, pos, entity);
+    }
+
+    @Nonnull
+    @SuppressWarnings("deprecation")
+    @Override
+    public InteractionResult use(@Nonnull BlockState state, @Nonnull Level level, @Nonnull BlockPos pos, @Nonnull Player player, @Nonnull InteractionHand hand, @Nonnull BlockHitResult hit) {
+        if (!level.isClientSide) {
```

Comment:
The method must return matching results on client and server or stuff may desync. First check for the item, then for `isClientSide` and if the correct item was used, return `InteractionResult.sidedSuccess(level.isClientSide)`